File: csharp/Types/Names.cs
using Newtonsoft.Json;

namespace Authdog.Types
{
    /// <summary>
    /// User name information
    /// </summary>
    public class Names
    {
        [JsonProperty("id")]
        public string Id { get; set; } = string.Empty;

        [JsonProperty("formatted")]
        public string? Formatted { get; set; }

        [JsonProperty("familyName")]
        public string FamilyName { get; set; } = string.Empty;

        [JsonProperty("givenName")]
        public string GivenName { get; set; } = string.Empty;

        [JsonProperty("middleName")]
        public string? MiddleName { get; set; }

        [JsonProperty("honorificPrefix")]
        public string? HonorificPrefix { get; set; }

        [JsonProperty("honorificSuffix")]
        public string? HonorificSuffix { get; set; }
    }
}

File: php/src/Types/Names.php
<?php

namespace Authdog\Types;

/**
 * User name information
 */
class Names
{
    public string $id;
    public ?string $formatted;
    public string $familyName;
    public string $givenName;
    public ?string $middleName;
    public ?string $honorificPrefix;
    public ?string $honorificSuffix;

    public function __construct(array $data)
    {
        $this->id = $data['id'] ?? '';
        $this->formatted = $data['formatted'] ?? null;
        $this->familyName = $data['familyName'] ?? '';
        $this->givenName = $data['givenName'] ?? '';
        $this->middleName = $data['middleName'] ?? null;
        $this->honorificPrefix = $data['honorificPrefix'] ?? null;
        $this->honorificSuffix = $data['honorificSuffix'] ?? null;
    }
}
